prov/usnic/src/usdf_ep_msg.c

```diff
@@ -175,8 +175,14 @@ int usdf_msg_fill_dom_attr(uint32_t version, struct fi_info *hints,
 	if (ret < 0)
 		return -FI_ENODATA;
 
-	if (!hints || !hints->domain_attr)
+	if (!hints || !hints->domain_attr) {
+		/* mr_mode behavior changed in version 1.5 from single flag to mode bits.
+		 * This is backward compatibility for Open MPI, which never give mr_mode flag.
```

Comment:
Note that Open MPI doesn't use RDM or MSG, so the "backwards compat" comment is a bit superfluous in the MSG / RDM cases.  Perhaps the comment could be:

```c
/* mr_mode behavior changed in version 1.5 from a single flag to mode bits.
 * Hence, if a version less than v1.5 was requested, return the prior default:
 * FI_MR_BASIC. */
```